src/EditorFeatures/Test2/IntelliSense/SignatureHelpControllerTests.vb

```diff
@@ -98,6 +98,18 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
             GetMocks(controller).PresenterSession.Verify(Sub(p) p.SelectNextItem(), Times.Once)
         End Sub
 
+        <WorkItem(1181, "https://github.com/dotnet/roslyn/issues/1181")>
+        <Fact>
+        Public Sub UpAndDownKeysShouldStillNavigateWhenDuplicateItemsAreFiltered()
+            Dim item = CreateItems(1).Single()
+            Dim controller = CreateController(items:={item, item}, waitForPresentation:=True)
```

Comment:
Digging in it gets a little more interesting.  Using `Items:=CreateItems(2)`, the set is kept at 2 items because `CreateItems()` creates simple `SignatureHelpItem`s and that class doesn't provide any sort of equality comparison; using the implementation here, `object.ReferenceEquals()` will return true because they're the same item; the manual repro from the bug causes `SymbolKeySignatureHelpItem` items to be created which _does_ override `bool Equals(object)`.  Hence the desired behavior with the manual repro as well as `items:={item, item}`, but not with `items:=CreateItems(2)`.

This leads us back to an earlier question: should `SignatureHelpItem` override `.Equals()`?  What does it mean to be equal?  Or should I update the test to use `SymbolKeySignatureHelpItem`?